app/javascript/controllers/mappings_controller.js

```diff
@@ -0,0 +1,297 @@
+    const textOntology = node.append("text")
+        .attr("dy", ".35em")
+        .style("text-anchor", "middle")
+        .style("font-size", "16px")
+        .style("fill", "white")
+        .style("font-weight", "600")
+        .text(d => (d.r > d.data.ontology_name.length * 5 && d.r > 20) ? d.data.ontology_name : "");
+
+    // Display number of mappings in 12px white below ontology names
+    const textMappings = node.append("text")
+        .attr("dy", "1.5em")
+        .style("text-anchor", "middle")
+        .style("font-size", "12px")
+        .style("fill", "white")
+        .text(d => (d.r > d.data.ontology_name.length * 5 && d.r > 20) ? d.data.ontology_mappings : "");
+
+    // Display ontology names in bubble tooltips on hover
+    circle.on("mouseover", function (event, d) {
+      if (!(d.r > d.data.ontology_name.length * 5 && d.r > 20)) {
+          // Remove existing tooltip
+          d3.selectAll(".bubble-tooltip").remove();
+
+          // Calculate tooltip position based on mouse coordinates
+          const tooltip = d3.select("body")
+              .append("div")
+              .attr("class", "bubble-tooltip")
+              .style("left", `${event.pageX + 10}px`) // Adjust position relative to mouse pointer
+              .style("top", `${event.pageY + 10}px`) // Adjust position relative to mouse pointer
+              .html(`<strong>${d.data.ontology_name}</strong><br>${d.data.ontology_mappings}`);
+      }
+    }).on("mouseout", function (event, d) {
+        // Remove tooltip on mouseout
+        d3.selectAll(".bubble-tooltip").remove();
+    });
+
+
+    this.svg = svg;
+  }
+  #hash_to_list(data){
+    return Object.keys(data).map(key => ({
+      ontology_name: this.#getLastPartOfUrl(key),
+      ontology_mappings: data[key],
+    }));
+  }
+  #getLastPartOfUrl(url) {
+    var parts = url.split('/');
+    return parts[parts.length - 1];
+  }
+  #center_scroll(frame){
+    frame.scrollTop = frame.scrollHeight / 2 - frame.clientHeight / 2;
+    frame.scrollLeft = frame.scrollWidth / 2 - frame.clientWidth / 2;
+  }
+  #normalization_ratio(ontologies_hash){
+    let maxValue = -Infinity
+    for (const value of Object.values(ontologies_hash)) {
+      maxValue = (maxValue < value) ? value : maxValue
+    }
+    let normalization_ratio = 1
+    while((maxValue / normalization_ratio)>10){
+      normalization_ratio *= 10
+    }
+    return normalization_ratio
+  }
+  #init_select(acronym){
+    let input = this.selectorTarget.querySelector('input')
+    input.value = acronym
+    input.dispatchEvent(new Event('input', { bubbles: true }));
+    for(let i = 0; i<this.selectorTarget.querySelectorAll('.option').length; i++){
+      const selectValue = this.selectorTarget.querySelectorAll('.option')[i]
+      const selectAcronym = this.#get_acronym(selectValue.getAttribute('data-value'))
+      if(selectAcronym == acronym){
+        selectValue.click()
+      }
+    }
+  }
+  #get_acronym(selectValue){
```

Comment:
can you not pass the acronym directly to the selected option as a data-attribute, so that you don't need all of this? that can break easily if you change selector text